include/utility/Logger.hxx

```diff
@@ -1,12 +1,96 @@
+                FormatContext &ctx) -> decltype(ctx.out()) {
+        // Directly use the formatter for std::string on TString's value
+        return fmt::formatter<std::string>::format((std::string)tstring, ctx);
+    }
+};
+
+// Specialize fmt::formatter for RVecs
+template <typename T> struct fmt::formatter<ROOT::VecOps::RVec<T>> {
+    // Parse is required by the fmt library. In this case, we don't need to
+    // parse anything specific.
+    constexpr auto parse(format_parse_context &ctx) -> decltype(ctx.begin()) {
+        return ctx.begin();
+    }
+
+    // Implement the format method to define how to convert
+    // ROOT::VecOps::RVec<T> to a string
+    template <typename FormatContext>
+    constexpr auto format(const ROOT::VecOps::RVec<T> &vec,
+                          FormatContext &ctx) -> decltype(ctx.out()) {
+        // Start with an opening bracket
+        fmt::format_to(ctx.out(), "[");
+        // Format each element separated by a comma
+        for (size_t i = 0; i < vec.size(); ++i) {
+            if (i > 0) {
+                fmt::format_to(ctx.out(), ", ");
+            }
+            fmt::format_to(ctx.out(), "{}", vec[i]);
+        }
+        // End with a closing bracket
+        return fmt::format_to(ctx.out(), "]\n");
+    }
+};
+
+// specialization for fmt::formatter for ROOT::Math::PtEtaPhiMVector
+template <> struct fmt::formatter<ROOT::Math::PtEtaPhiMVector> {
+    // Parse is required by the fmt library. In this case, we don't need to
+    // parse anything specific.
+    constexpr auto parse(format_parse_context &ctx) -> decltype(ctx.begin()) {
+        return ctx.begin();
+    }
+
+    // Implement the format method to define how to convert
+    // ROOT::Math::PtEtaPhiMVector to a string
+    template <typename FormatContext>
+    constexpr auto format(const ROOT::Math::PtEtaPhiMVector &vec,
+                          FormatContext &ctx) -> decltype(ctx.out()) {
+        // Format the vector as needed, for example:
+        return fmt::format_to(ctx.out(), "[Pt: {}, Eta: {}, Phi: {}, M: {}]\n",
+                              vec.Pt(), vec.Eta(), vec.Phi(), vec.M());
+    }
+};
+
+// Generic formatter function for std::bitset of any size
+template <std::size_t N> struct BitsetFormatter {
+    static auto format(fmt::format_context &ctx,
+                       const std::bitset<N> &b) -> decltype(ctx.out()) {
+        fmt::format_to(ctx.out(), "[");
+        for (size_t i = 0; i < b.size(); ++i) {
+            fmt::format_to(ctx.out(), "{}", b.test(i) ? '1' : '0');
+        }
+        return fmt::format_to(ctx.out(), "]\n");
+    }
+};
+
+// Register the formatter with fmt library for any size of std::bitset
+template <std::size_t N>
+struct fmt::formatter<std::bitset<N>> : formatter<std::string> {
+    template <typename FormatContext>
+    auto format(const std::bitset<N> &b,
+                FormatContext &ctx) -> decltype(ctx.out()) {
+        return BitsetFormatter<N>::format(ctx, b);
+    }
+};
+
```

Comment:
Does this already cover all the cases that we have in Crown?

Reply:
most basic types are covered, only the ones that are not "common" like the ROOT special types. At least the tau analysis does compile with this. Maybe other analyses need additional types here